unidecode/__init__.py

```diff
@@ -91,13 +102,24 @@ def _unidecode(string):
         except KeyError:
             try:
                 mod = __import__('unidecode.x%03x'%(section), globals(), locals(), ['data'])
-            except ImportError:
-                Cache[section] = None
-                continue   # No match: ignore this character and carry on.
-
-            Cache[section] = table = mod.data
-
-        if table and len(table) > position:
-            retval.append( table[position] )
+                Cache[section] = table = mod.data
+            except ImportError as e:
+                Cache[section] = table = None
+                if errors=='strict':
+                    raise e
+
+        # set return value to the found charace
+        if table and len(table) > position and table[position] != '' :
```

Comment:
This code assumes that if the transliteration table contains an empty string, it means there is no transliteration data about this character. This isn't true. Some characters are purposely replaced with an empty string (e.g. some zero width spaces like U+2060, etc.). On the other hand, some characters missing from the tables have `[?]` in the tables.

The `[?]` cases and the fact that it isn't possible to distinguish between the "no data" and "replace with empty string" cases is a bug in Unidecode.